apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.UnitTests/PriorityListTests.cs

```diff
@@ -0,0 +1,235 @@
+
+            priorityList.Add(new TestItem(3));
+            priorityList.Add(new TestItem(11));
+            priorityList.Add(new TestItem(2));
+            priorityList.Add(new TestItem(0));
+            priorityList.Add(new TestItem(6));
+
+            AssertSorted(priorityList);
+
+            Assert.Equal(0, priorityList.GetTop().Priority);
+        }
+
+        [Fact]
+        public void RemoveLast()
+        {
+            PriorityList<TestItem> priorityList = new PriorityList<TestItem>();
+            TestItem itemToRemove;
+
+            priorityList.Add(new TestItem(3));
+            priorityList.Add(itemToRemove = new TestItem(11));
+            priorityList.Add(new TestItem(2));
+            priorityList.Add(new TestItem(0));
+            priorityList.Add(new TestItem(6));
+
+            priorityList.RemoveLast();
+            AssertSorted(priorityList);
+            AssertNotExist(priorityList, itemToRemove);
+        }
+
+        [Fact]
+        public void PriorityChangeResortsList()
+        {
+            PriorityList<TestItem> priorityList = new PriorityList<TestItem>();
+            TestItem changingItem;
+
+            priorityList.Add(new TestItem(3));
+            priorityList.Add(new TestItem(11));
+            priorityList.Add(changingItem = new TestItem(2));
+            priorityList.Add(new TestItem(0));
+            priorityList.Add(new TestItem(6));
+            AssertSorted(priorityList);
+            changingItem.Priority = 20;
+            AssertNotSorted(priorityList);
+            changingItem.FirePriorityChanged();
+            AssertSorted(priorityList);
+        }
+
+        private void AssertExist(PriorityList<TestItem> priorityList, params TestItem[] testItems)
+        {
+            foreach (var testItem in testItems)
+            {
+                Assert.True(priorityList.TryFindLinear(x => ReferenceEquals(x, testItem), out _));
+            }
+        }
+
+        private void AssertNotExist(PriorityList<TestItem> priorityList, params TestItem[] testItems)
+        {
+            foreach (var testItem in testItems)
+            {
+                Assert.False(priorityList.TryFindLinear(x => ReferenceEquals(x, testItem), out _));
+            }
+        }
+
+        private void AssertSorted(PriorityList<TestItem> priorityList)
+        {
+            Assert.True(IsSorted(priorityList));
+        }
+        private void AssertNotSorted(PriorityList<TestItem> priorityList)
+        {
+            Assert.False(IsSorted(priorityList));
+        }
+        private bool IsSorted(PriorityList<TestItem> priorityList)
+        {
+            int lastPriority = 0;
+            foreach (var testItem in priorityList.GetSnapshot())
+            {
+                if (lastPriority > testItem.Priority)
+                {
+                    return false;
+                }
+                lastPriority = testItem.Priority;
+            }
+            return true;
```

Comment:
I'd say it's clearer precisely *because* PriorityList uses the fact that IPriorityListItem is comparable. The "coincidence" is that TestItem implements `IComparable<TestItem>` by comparing Priority properties. So the check would be more general IMO.

(And yes, it does an unnecessary sort, but that's trivial amounts of extra work for these tests.)

I'm not massively bothered either way though - not a merge-blocker for me.